django_kafka/retry/topic.py

```diff
@@ -0,0 +1,72 @@
+import re
+
+from confluent_kafka import cimpl
+
+from django_kafka.exceptions import DjangoKafkaError
+from django_kafka.retry.headers import RetryHeader
+from django_kafka.serialization import NoOpSerializer
+from django_kafka.topic import Topic
+
+
+class RetryTopic(Topic):
+    key_serializer = NoOpSerializer
+    value_serializer = NoOpSerializer
+
+    key_deserializer = NoOpSerializer
+    value_deserializer = NoOpSerializer
+
+    def __init__(self, group_id: str, main_topic: Topic):
+        if not main_topic.retry:
+            raise DjangoKafkaError(f"Topic {main_topic} is not marked for retry")
+        self.main_topic = main_topic
+        self.group_id = group_id
+
+    @property
+    def name(self) -> str:
+        group_id = re.escape(self.group_id)
+        if self.main_topic.is_regex():
+            topic_name = f"({self.main_topic.name.lstrip('^').rstrip('$')})"
+        else:
+            topic_name = re.escape(self.main_topic.name)
+
+        return rf"^{group_id}\.{topic_name}\.retry\.[0-9]+$"
+
+    @classmethod
+    def get_attempt(cls, topic_name: str) -> int:
+        match = re.search(r"\.retry\.([0-9]+)$", topic_name)
+        return int(match.group(1)) if match else 0
+
+    def get_produce_name(self, topic_name: str, attempt: int) -> str:
```

Comment:
Depending on the attempt, `msg.topic()` can be different. For attempt 1, the `msg.topic()` is the original topic. For greater, `msg.topic()` is the retry topic (`{self.group_id}.{topic_name}.retry.{attempt}`). So depending on the attempt, I have to append `retry.1`, or regex sub the attempt numbers.

I have access to the main topic, but I can't do:

`return {self.group_id}.{self.main_topic.name}.retry.{attempt}`

Because `self.main_topic.name` may be regex hitting multiple topics, whereas I need a concrete topic name, not a regex.

I think maybe we should rename the `name` property because it gives the impression it's a concrete name and not possibly a regex. In Confluent Kafka it's just called `topic`. But if I put a `topic` property in the `Topic` class, we'd be accessing `topic.topic` a lot.